### PR TITLE
Geometry Editor

### PR DESCRIPTION
**Pull request by williambohrmann3**
# Description

- Add new sample: Create and edit geometries
- Feature Create and edit geometries
- Remove sample: Sketch on map
- Remove Sketch on map resources from .Net.csproj and .NetFramework.csproj

## Type of change

- New sample implementation

## Platforms tested on

- [x] WPF .NET 6
- [x] WPF Framework
- [x] WinUI
- [x] MAUI WinUI
- [x] MAUI Android
- [x] MAUI iOS
- [x] MAUI MacCatalyst

## Checklist

- [x] Self-review of changes
- [x] All changes work as expected on all affected platforms
- [x] There are no warnings related to changes
- [x] Code is commented and follows .NET conventions and standards
- [x] Codemaid and XAML styler extensions have been run on every changed file
- [x] No unrelated changes have been made to any other code or project files
- [x] Screenshots are correct size and display in description tab (800 x 600 on Windows, MAUI mobile platforms use the MAUI Windows screenshot)

### CODE REVIEW

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,421 @@
...
+                if (!_geometryEditor.IsStarted)
+                {
+                    // Identify graphics in the graphics overlay using the mouse point.
+                    IReadOnlyList<IdentifyGraphicsOverlayResult> results = await MyMapView.IdentifyGraphicsOverlaysAsync(e.Position, 2, false);
+
+                    // Get the first graphic if results were found.
+                    IdentifyGraphicsOverlayResult idResult = results.FirstOrDefault();
+                    if (idResult != null && idResult.Graphics.Count > 0)
+                    {
+                        _selectedGraphic = idResult.Graphics.FirstOrDefault();
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:370]: Since you do a prior check for count, I think this can use index `idResult.Graphics[0]` or if you want even simplify that
```CSharp
if (MyMapView.GeometryEditor.IsStarted) return; // editing already in progress

_selectedGraphic = results.FirstOrDefault()?.Graphics?.FirstOrDefault();

if(_selectedGraphic is null)  return; // nothing to edit
```

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+                                break;
+                        }
+
+                        // Hide the selected graphic and start an editing session with a copy of it.
+                        _geometryEditor.Start(_selectedGraphic.Geometry);
+                        _selectedGraphic.IsVisible = false;
+                    }
+                    else
+                    {
+                        _selectedGraphic = null;
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:410]: Do you ever need to reset here or only on Save/Discard?

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+                    }
+                }
+            }
+            catch (Exception ex)
+            {
+                // Report exceptions.
+                MessageBox.Show("Editing error: " + ex.Message, ex.GetType().Name, MessageBoxButton.OK, MessageBoxImage.Error);
+            }
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:418]: Will you need to restore the selection and visibility of `_selectedGraphic` if there's any on error?

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+                       Content="Create"
+                       FontSize="20"
+                       FontWeight="Bold" />
+                <Button x:Name="PointButton"
+                        Grid.Row="1"
+                        Click="PointButton_Click">
+                    <Button.ContentTemplate>
+                        <DataTemplate>
+                            <StackPanel>
+                                <TextBlock HorizontalAlignment="Center"
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml:41]: When you find several properties tend to repeat, I think it's good indication to introduce a style.

```XML
    <UserControl.Resources>
        <Style x:Key="IconStyle"
               TargetType="TextBlock">
            <Style.Setters>
                <Setter Property="HorizontalAlignment"
                        Value="Center" />
                <Setter Property="FontFamily"
                        Value="pack://application:,,,/Resources/Fonts/#calcite-ui-icons-24" />
                <Setter Property="FontSize"
                        Value="40" />
            </Style.Setters>
        </Style>
    </UserControl.Resources>

    <!-- more code here -->
                                <TextBlock Style="{StaticResource IconStyle}"
                                           Text="&#xe21c;" />

 
```

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 5]
```diff
@@ -0,0 +1,421 @@
...
+            else
+            {
+                _geometryCreated = true;
+
+                // Get the geometry from the geometry editor and create a new graphic for it.
+                Geometry geometry = _geometryEditor.Stop();
+                var graphic = new Graphic(geometry);
+
+                // Set the graphic style based on geometry type.
+                switch (geometry.GeometryType)
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:272]: I wonder if this can be refactored into a `GetSymbol(..)` method so that it does not look like you have to do a lot to add a new graphic from the editor. I can't remember if there's a chance geometry result is null, maybe a null-check before getting it's type and adding to the overlay?

```CSharp
var geometry = MyMapView.GeometryEditor.Stop();
var symbol = GetSymbol(geometry);
MyMapView.GraphicsOverlay[0].Graphics.Add(new Graphic(geometry, symbol));
```

I think the if/else can even be combined a little.

```CSharp
var geometry = MyMapView.GeometryEditor.Stop();
if(_selectedGraphic is not null)
{
  _selectedGraphic.Geometry = geometry;
  ResetSelectedGraphic();
}
else
{
  var symbol = GetSymbol(geometry);
   MyMapView.GraphicsOverlay[0].Graphics.Add(new Graphic(geometry, symbol));
}
```

You may need to reset when identify fails, save, discard, no?
```CSharp
        private void ResetSelectedGraphic()
        {
            if (_selectedGraphic is not null)
            {
                _selectedGraphic.IsSelected = false;
                _selectedGraphic.IsVisible = true;
            }
            _selectedGraphic = null;
        }
```

**Review by jnery — CHANGES_REQUESTED** [review]
Great job, @williambohrmann3! All samples are looking very polished.

I just reviewed the WPF bits but most of my suggestions probably apply to other platforms. Please feel free to apply only what you think is necessary. Like if samples team have a prescribed way of doing things, then go with consistency. I think most of my comments are for readability or making things concise.

**Comment by duffh** [comment]
Missing nuget package updates to 200.2

**Comment by duffh** [comment]
Slightly unintuitive UI when you can select the `Freehand Tool` but then `Point` and `Multipoint` change the tool back to `Vertex Tool`. Maybe disable these buttons while `Freehand Tool` is selected?

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/readme.md [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+
+* Geometry
+* GeometryEditor
+* Graphic
+* GraphicsOverlay
+* MapView
+
+## Additional information
+
+The sample opens with the ArcGIS Imagery basemap centered on the island of Inis Me�in (Aran Islands) in Ireland. Inis Me�in comprises a landscape of interlinked stone walls, roads, buildings, archaeological sites, and geological features, producing complex geometrical relationships.
```
Comment (duffh) [inline src/WinUI/ArcGIS.WinUI.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/readme.md:47]: The `á` character is not rendering properly in the sample viewer.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml [hunk 7]
```diff
@@ -0,0 +1,224 @@
+<UserControl x:Class="ArcGIS.WinUI.Samples.CreateAndEditGeometries.CreateAndEditGeometries"
+             xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+             xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+             xmlns:esriUI="using:Esri.ArcGISRuntime.UI.Controls">
+    <UserControl.Resources>
+        <FontFamily x:Key="Calcite">/Resources/Fonts/calcite-ui-icons-24.ttf#calcite-ui-icons-24</FontFamily>
+    </UserControl.Resources>
+    <Grid>
```
Comment (duffh) [inline src/WinUI/ArcGIS.WinUI.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml:8]: UI looks a little cramped and maybe a little too large.

**Review by duffh — CHANGES_REQUESTED** [review]
Looks good, just a few comments just now.

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 8]
```diff
@@ -0,0 +1,421 @@
...
+            // Yellow circle for multipoints.
+            _multiPointSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Circle, Color.Yellow, 5);
+            // Thin blue line for polylines.
+            _lineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Solid, Color.Blue, 2);
+            // Black outline for polygons.
+            var polygonLineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Dash, Color.Black, 1);
+            // Translucent red interior for polygons.
+            _fillSymbol = new SimpleFillSymbol(SimpleFillSymbolStyle.Solid, Color.FromArgb(70, 255, 0, 0), polygonLineSymbol);
+
+            // Create geometry objects from JSON formatted strings.
```
[the remark this answers is not in the document]
Comment (duffh) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:94]: Another idea is to move this json to a readonly member variable outside of the method scope to improve readability.

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml (hunk 4 above)
Comment (duffh) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml:41]: Agree with Jennifer, I think the xaml just now is slightly too verbose and hard to follow.

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 9]
```diff
@@ -0,0 +1,421 @@
...
+        tags: new[] { "draw", "edit", "freehand", "geometry editor", "sketch", "vertex" })]
+    public partial class CreateAndEditGeometries
+    {
+        private GeometryEditor _geometryEditor;
+        private SimpleFillSymbol _fillSymbol;
+        private SimpleLineSymbol _lineSymbol;
+        private SimpleMarkerSymbol _pointSymbol, _multiPointSymbol;
+        private List<Button> _geometryButtons;
+        private Graphic _selectedGraphic;
+        private GraphicsOverlay _graphicsOverlay;
```
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:42]: Yeah the `GraphicsOverlay` and `GeometryEditor` members aren't technically necessary. They're in this sample for readability and SDK parity purposes.

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 10]
```diff
@@ -0,0 +1,421 @@
...
+        public CreateAndEditGeometries()
+        {
+            InitializeComponent();
+            Initialize();
+        }
+
+        private void Initialize()
+        {
+            // Create a map with standard imagery basemap style.
+            var map = new Map(BasemapStyle.ArcGISImageryStandard);
```
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:54]: I've collapsed the map constructor code.

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 11]
```diff
@@ -0,0 +1,421 @@
...
+            MyMapView.GeometryEditor = _geometryEditor;
+
+            // Create vertex and freehand tools for the geometry editor and add to the combo box.
+            ToolComboBox.ItemsSource = new Dictionary<string, GeometryEditorTool>
+            {
+                {"Vertex Tool", new VertexTool()}, {"Freehand Tool", new FreehandTool()}
+            };
+
+            // Display the tool name in the combo box.
+            ToolComboBox.DisplayMemberPath = "Key";
```
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:77]: Not needed in codebehind - moved this to XAML

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 12]
```diff
@@ -0,0 +1,421 @@
...
+        #endregion Enable and disable geometry buttons methods
+
+        private async void MyMapView_GeoViewTapped(object sender, GeoViewInputEventArgs e)
+        {
+            try
+            {
+                if (!_geometryEditor.IsStarted)
+                {
+                    // Identify graphics in the graphics overlay using the mouse point.
+                    IReadOnlyList<IdentifyGraphicsOverlayResult> results = await MyMapView.IdentifyGraphicsOverlaysAsync(e.Position, 2, false);
```
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:364]: Good catch, tolerance should be 5. I got 2 from sketch-on-map sample.

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 13]
```diff
@@ -0,0 +1,421 @@
...
+    public partial class CreateAndEditGeometries
+    {
+        private GeometryEditor _geometryEditor;
+        private SimpleFillSymbol _fillSymbol;
+        private SimpleLineSymbol _lineSymbol;
+        private SimpleMarkerSymbol _pointSymbol, _multiPointSymbol;
+        private List<Button> _geometryButtons;
+        private Graphic _selectedGraphic;
+        private GraphicsOverlay _graphicsOverlay;
+        private bool _geometryCreated = true;
```
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:43]: A consequence of removing the flag and creating this new event is now the delete all button is enabled when the geometry editor is started. I don't think this is a problem - stopping the editor and ensuring the _selectedGraphic is null remedies any bugs. This is actually quite nice - say the user is editing or creating a new geometry and decides to start from scratch - they can now do that without clicking the stop and discard edits button.

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs (hunk 3 above)
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:418]: I wonder how risky it would be to get rid of this try-catch block. I haven't gotten exceptions after thorough testing all buttons.

Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:418]: Disregard previous comment - would need to surround the await code and anything that uses the results in a try/catch block. And yes, used your `ResetSelectedGraphic()` method in this case.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/readme.md (hunk 6 above)
Comment (williambohrmann3) [reply src/WinUI/ArcGIS.WinUI.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/readme.md:47]: Maybe we change the text to Inishmaan unless you know of a workaround?

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 14]
```diff
@@ -0,0 +1,421 @@
...
+                _graphicsOverlay.Graphics.Add(graphic);
+            }
+
+            // Create a list of geometry buttons for easy enable and disable.
+            _geometryButtons = new List<Button>
+            {
+                PointButton, MultipointButton, PolylineButton, PolygonButton
+            };
+
+            // Update the UI to reflect geometry editor property changes.
```
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:121]: In an effort to reduce code, I binded SaveButton to CanUndo. Although there are instances as described above where there is nothing to save, the logic required to consider save button conditions for different geometry types and how the editor was started is beyond the scope of this sample. In my eyes, this is miscommunication to the user, but in doing this, no bugs are introduce and it makes the code more readable.

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs (hunk 13 above)
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:43]: While simplifying my UI logic, the delete all button is now binded to the inverse of GeometryEditor.IsStarted. This doesn't consider the amount of geometries enabled, so when the geometry editor is stopped and there are no graphics, this button is still enabled. A minor UI miscommunication, but nothing buggy.

File: src/WinUI/ArcGIS.WinUI.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/readme.md (hunk 6 above)
Comment (duffh) [reply src/WinUI/ArcGIS.WinUI.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/readme.md:47]: I think for this case just use `a` rather than `á` it's close enough for our purposes.

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 15]
```diff
@@ -250,172 +180,183 @@ private void DeleteSelectedButton_Click(object sender, RoutedEventArgs e)
...
+                // Try to get the first graphic from the first result.
+                _selectedGraphic = results.FirstOrDefault()?.Graphics?.FirstOrDefault();
+            }
+            catch (Exception ex)
+            {
+                // Report exceptions.
+                MessageBox.Show("Editing error: " + ex.Message, ex.GetType().Name, MessageBoxButton.OK, MessageBoxImage.Error);
+
+                ResetFromEditingSession();
+                return;
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:231]: do you need `return` or will the reset already make _selectedGraphic null?

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 16]
```diff
@@ -250,172 +180,183 @@ private void DeleteSelectedButton_Click(object sender, RoutedEventArgs e)
...
+                // Identify graphics in the graphics overlay using the mouse point.
+                IReadOnlyList<IdentifyGraphicsOverlayResult> results = await MyMapView.IdentifyGraphicsOverlaysAsync(e.Position, 5, false);
+
+                // Try to get the first graphic from the first result.
+                _selectedGraphic = results.FirstOrDefault()?.Graphics?.FirstOrDefault();
+            }
+            catch (Exception ex)
+            {
+                // Report exceptions.
+                MessageBox.Show("Editing error: " + ex.Message, ex.GetType().Name, MessageBoxButton.OK, MessageBoxImage.Error);
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:228]: I don't know if there's a precedence in the sample already on how message is displayed but I don't think this will be an "editing error" but an error in the identify operation.

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 17]
```diff
@@ -250,172 +180,183 @@ private void DeleteSelectedButton_Click(object sender, RoutedEventArgs e)
...
+            // Return since no graphic was selected.
+            if (_selectedGraphic == null) return;
+
+            _selectedGraphic.IsSelected = true;
+
+            // Configure the UI depending on the geometry type.
+            GeometryType geometryType = _selectedGraphic.Geometry.GeometryType;
+            if (geometryType == GeometryType.Point || geometryType == GeometryType.Multipoint)
+            {
+                ToolComboBox.SelectedIndex = 0;
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:243]: I think we want VertexTool to always be the default.
Allowing users to change it to Freehand when Polyline/Polygon seem enough. Which you're doing by handling the `ComboBox.SelectionChanged` event

**Review by jnery — CHANGES_REQUESTED** [review]
Thanks for the update, just few more questions; but otherwise, I think it's much closer to done!

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 18]
```diff
@@ -0,0 +1,362 @@
...
+        // Delete the currently selected element of the geometry editor.
+        private void DeleteSelectedButton_Click(object sender, RoutedEventArgs e)
+        {
+            _geometryEditor.DeleteSelectedElement();
+        }
+
+        // Update an existing graphic or create a new graphic.
+        private void SaveButton_Click(object sender, RoutedEventArgs e)
+        {
+            Geometry geometry = _geometryEditor.Stop();
```
Comment (jnery) [inline src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:183]: Actually Stop() could return a null geometry, maybe add a check? Otherwise, you get null ref below for getting its type to apply an appropriate symbol.

**Review by jnery — CHANGES_REQUESTED** [review]

File: src/MAUI/Maui.Samples/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 19]
```diff
@@ -0,0 +1,355 @@
...
+        {
+            // Create symbols for displaying new geometries.
+            _pointSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Square, Color.OrangeRed, 10);
+            _multiPointSymbol = new SimpleMarkerSymbol(SimpleMarkerSymbolStyle.Circle, Color.Yellow, 5);
+            _polylineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Solid, Color.Blue, 2);
+            var polygonLineSymbol = new SimpleLineSymbol(SimpleLineSymbolStyle.Dash, Color.Black, 1);
+            _polygonSymbol = new SimpleFillSymbol(SimpleFillSymbolStyle.Solid, Color.FromArgb(70, 255, 0, 0), polygonLineSymbol);
+
+            // Create geometry objects from JSON formatted strings.
+            var houseCoordinates = (MapPoint)Geometry.FromJson
```
Comment (duffh) [inline src/MAUI/Maui.Samples/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:259]: I'd pull these json strings out of the method as constant strings, it just tidies up readability. We do this in other samples.

**Review by duffh — CHANGES_REQUESTED** [review]
Looking better!

File: src/WPF/WPF.Viewer/Converters/BoolNegationConverter.cs [hunk 20]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+using System.Windows.Data;
+
+namespace ArcGIS.WPF.Viewer.Converters
+{
```
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Converters/BoolNegationConverter.cs:5]: [BoolNegationConverter](https://learn.microsoft.com/en-us/dotnet/api/microsoft.toolkit.uwp.ui.converters.boolnegationconverter?view=win-comm-toolkit-dotnet-7.1l) is what they named this converter in UWP toolkit and we're using that one in WinUI implementation of this sample. Added a comment.

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 21]
```diff
@@ -222,10 +147,15 @@ private void PolygonButton_Click(object sender, RoutedEventArgs e)
         private void ToolComboBox_SelectionChanged(object sender, SelectionChangedEventArgs e)
         {
             var keyValuePair = (KeyValuePair<string, GeometryEditorTool>)ToolComboBox.SelectedItem;
-            _geometryEditor.Tool = keyValuePair.Value;
+            var tool = (GeometryEditorTool)keyValuePair.Value;
+            _geometryEditor.Tool = tool;
+
+            // Account for case when vertex tool is selected and geometry editor is started with a polyline or polygon geometry type.
+            // Ensure point and multipoint buttons are only enabled when the selected tool is a vertex tool.
+            PointButton.IsEnabled = MultipointButton.IsEnabled = !_geometryEditor.IsStarted && tool is VertexTool;
```
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:155]: I don't believe this is technically necessary. But is it worth adding a geometry editor type conditional check around this?

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs [hunk 22]
```diff
@@ -250,172 +180,183 @@ private void DeleteSelectedButton_Click(object sender, RoutedEventArgs e)
...
-                button.IsEnabled = false;
+                _selectedGraphic.IsSelected = false;
+                _selectedGraphic.IsVisible = true;
             }
-        }
+            _selectedGraphic = null;
 
-        #endregion Enable and disable geometry buttons methods
+            // Point and multipoint sessions do not support the vertex tool.
+            PointButton.IsEnabled = MultipointButton.IsEnabled = _geometryEditor.Tool is VertexTool;
```
[the remark this answers is not in the document]
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:326]: This change was made because of @duffh left this [comment](https://github.com/Esri/arcgis-maps-sdk-dotnet-samples/pull/1251#issuecomment-1578288015)

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs (hunk 15 above)
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:231]: nice catch!

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs (hunk 16 above)
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:228]: Will make it display as "error"

File: src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs (hunk 17 above)
Comment (williambohrmann3) [reply src/WPF/WPF.Viewer/Samples/GraphicsOverlay/CreateAndEditGeometries/CreateAndEditGeometries.xaml.cs:243]: If user has freehand tool enabled, selects a polyline/polygon, no need to change the tool to vertex tool. But if they have freehand tool enabled, select a point/multipoint, it is necessary to change the tool to a vertex tool.

**Review by jnery — APPROVED** [review]
LGTM, thanks 👍

**Comment by williambohrmann3** [comment]
Checks are failing because of lowercase screenshot name for MAUI.

**Comment by duffh** [comment]
> Checks are failing because of lowercase screenshot name for MAUI.

Looks like the README check is failing as the metadata is inconsistent, the difference is that I think it is expecting a .png image but we are using a .jpg for our screenshot. I'll take a look at the check.

**Comment by duffh** [comment]
The checks are failing because the screenshot should be a `.jpg` file not a `.png`. If you update this both checks should pass.

**Review by duffh — APPROVED** [review]
